### ISSUE TITLE
Inconsistent representation of MultitaskMultivariateNormal covariance

### ISSUE THREAD

**Issue by Balandat** [cornellius-gp/gpytorch#537]
Consider a 2-task model with 3 points each. The shape of the joint covariance matrix `C` is `6x6`, but the current code seems to assume different structures in different places.  

1. `from_independent_mvns` assumes that `C` has block structure w.r.t. the tasks, i.e. that `C = [[C_a, 0], [0, C_b]]`, where `C_a` and `C_b` are the marginal covariances across the data points of task `a` and `b`, respectively.

2. `variance` and `rsample` assume that `C` has block structure w.r.t. the data points, i.e. that `C[2*i:2*(i+1), 2*i:2*(i+1)]`  is the covariance across tasks for data point `i`.

Clearly, the two are incompatible. This means that the variances and samples generated for a MTMVN generated from `from_independent_mvns` are incorrect.

**Comment by Balandat** [comment]
Here's a (terribly inefficient) function for converting from 1. to 2. for full (non-lazy) tensors.
```
def convert_matrix(task_block_diag_cov, n_tasks):
    n = task_block_diag_cov.shape[-1] // n_tasks
    cout = torch.zeros_like(task_block_diag_cov)
    for t in range(n_tasks):
        for i in range(n):
            for j in range(n):
                val = task_block_diag_cov[..., t*n+i, t*n+j]
                cout[..., t+n_tasks*i, t+n_tasks*j] = val
    return cout
```

**Comment by Balandat** [comment]
It seems like having the option of using both representations internally could make a lot of sense in order to exploit structure.

**Comment by Balandat** [comment]
Resolved via #538

**Comment by Balandat** [comment]
Actually, there is is a bug in #538 - working on a fix of the fix
